packages/docusaurus-theme-classic/src/theme/NavbarItem/index.tsx

```diff
@@ -65,5 +34,6 @@ export default function NavbarItem({type, ...props}: Props): JSX.Element {
     (props as DropdownNavbarItemProps).items !== undefined,
   );
   const NavbarItemComponent = getNavbarItemComponent(componentType);
+  // @ts-expect-error: how to type this?
   return <NavbarItemComponent {...props} />;
```

Comment:
```suggestion
  return <NavbarItemComponent {...(props as never)} />;
```

Will break in the next TS version, but that's something to worry about later. Polymorphic components are hard to type.